sapl/materia/views.py

```diff
@@ -480,44 +484,81 @@ def get_rows(self, object_list):
                     obj.data_recebimento = 'Não recebida'
                 else:
                     obj.data_recebimento = obj.data_recebimento.strftime(
-                                            "%d/%m/%Y %H:%M")
+                        "%d/%m/%Y %H:%M")
 
             return [self._as_row(obj) for obj in object_list]
 
         def get_queryset(self):
+            # Só tem acesso as Proposicoes criadas por ele que ainda nao foram
+            # recebidas ou foram devolvidas
             lista = Proposicao.objects.filter(
                 autor__user_id=self.request.user.id)
+            lista = lista.filter(
+                Q(data_recebimento__isnull=True) |
+                Q(data_devolucao__isnull=False))
+
             return lista
 
     class DeleteView(PermissionRequiredMixin, CrudDeleteView):
         permission_required = {'materia.delete_proposicao'}
 
+        def has_permission(self):
+            perms = self.get_permission_required()
+            if not self.request.user.has_perms(perms):
+                return False
+
+            return (Proposicao.objects.filter(
+                id=self.kwargs['pk'],
+                autor__user_id=self.request.user.id).exists())
+
         def delete(self, request, *args, **kwargs):
             proposicao = Proposicao.objects.get(id=self.kwargs['pk'])
 
-            if not proposicao.data_envio:
+            if not proposicao.data_envio or proposicao.data_devolucao:
                 proposicao.delete()
                 return HttpResponseRedirect(
                     reverse('sapl.materia:proposicao_list'))
-            else:
+
+            elif not proposicao.data_recebimento:
                 proposicao.data_envio = None
                 proposicao.save()
                 return HttpResponseRedirect(
                     reverse('sapl.materia:proposicao_detail',
                             kwargs={'pk': proposicao.pk}))
 
+            else:
+                msg = _('Essa proposição já foi recebida. ' +
+                        'Não pode mais ser excluída/recuperada')
+                messages.add_message(self.request, messages.ERROR, msg)
+                return HttpResponseRedirect(
+                    reverse('sapl.materia:proposicao_detail',
+                            kwargs={'pk': proposicao.pk}))
+
 
 class ReciboProposicaoView(TemplateView):
     template_name = "materia/recibo_proposicao.html"
+    permission_required = permissoes_autor()
+
+    def has_permission(self):
+            perms = self.get_permission_required()
+            if self.request.user.has_perms(perms):
```

Comment:
Aqui também dá pra usar o ``if not self.request.user.has_perms(perms): return False``